base/arraymath.jl

```diff
@@ -472,12 +469,16 @@ for (f, f!, fp, op) = ((:cumsum, :cumsum!, :cumsum_pairwise!, :+),
     @eval function ($f!)(result::AbstractVector, v::AbstractVector)
         n = length(v)
         if n == 0; return result; end
-        ($fp)(v, result, $(op==:+ ? :(zero(first(v))) : :(one(first(v)))), first(indices(v,1)), n)
+        li = linearindices(v)
```

Comment:
Ah, I see that the `length` documentation says it returns "For ordered, indexable collections, the maximum index i for which getindex(collection, i) is valid."

Okay, changed it to `length(li)` as you suggest.   It still seems strange to me.

(Note also that this `cumsum` method is only for `AbstractVector`, and the `length` documentation says that it "Returns the number of elements in A." for any `AbstractArray`.)